pkg/skuba/actions/node/upgrade/apply.go

```diff
@@ -128,6 +128,12 @@ func Apply(client clientset.Interface, target *deployments.Target) error {
 			return err
 		}
 	}
+	if nodeVersionInfoUpdate.HasMajorOrMinorUpdate() {
+		err = target.Apply(nil, "cri.configure")
```

Comment:
In the bootstrap and join cases, this target is guarded by a check for the addon config file: https://github.com/SUSE/skuba/blob/ecf11c5b5f5510fb2cd660dd196cbc9afc0f33a4/pkg/skuba/actions/node/bootstrap/bootstrap.go#L122-L126 is that not required here as well?

Reply:
Instead sending crio configuration file directly on the upgraded nodes, we should first take care about updating <CLUSTER-INIT>/addons/crio/conf.d because this is the place where user will look for any changes. If this would happen we don't have to this change at all, because PR https://github.com/SUSE/skuba/pull/1061 is relaying on configuration from <CLUSTER-INIT>

Reply:
> In the bootstrap and join cases, this target is guarded by a check for the addon config file:
> 
> https://github.com/SUSE/skuba/blob/ecf11c5b5f5510fb2cd660dd196cbc9afc0f33a4/pkg/skuba/actions/node/bootstrap/bootstrap.go#L122-L126
> is that not required here as well?

The file in sysconfig wasn't changed since its inception (until now, where we can remove it). So I guess the sysconfg doesn't need to run on upgrades. However you're right, we should probably have extra actions to behave in a consistent manner. For example, something to cleanup the file, if version of the upgrade is 17.

Reply:
> The file in sysconfig wasn't changed since its inception (until now, where we can remove it). So I guess the sysconfg doesn't need to run on upgrades. However you're right, we should probably have extra actions to behave in a consistent manner. For example, something to cleanup the file, if version of the upgrade is 17.

sysconfig file is written by the cri-o package everytime when it is upgraded

Reply:
But we won't need it starting from crio 1.18, isn't it? I thought this was removed from the package.

Reply:
cri-o `sysconfig` is still OS method for providing service configuration, for v1.18 we want to only stop using it

Reply:
That's the difference I didn't know. I expected it was removed from the package. Thanks for the clarification.

Reply:
I have added a comment in the code, I hope it clarifies.